src/ethereum/EthereumNetwork.ts

```diff
@@ -1502,6 +1514,88 @@ export class EthereumNetwork {
     }
   }
 
+  /**
+   * Check the eth-balance-checker contract for balances
+   */
+  // @ts-expect-error
+  async checkEthBalChecker(): Promise<EthereumNetworkUpdate> {
+    const { allTokensMap, networkInfo, walletLocalData, currencyInfo } =
+      this.ethEngine
+    const { chainParams, rpcServers, ethBalCheckerContract } = networkInfo
+
+    const tokenBal: { [currencyCode: string]: string } = {}
+    if (ethBalCheckerContract == null) return tokenBal
+
+    // Address for querying ETH balance on ETH network, MATIC on MATIC, etc.
+    const mainnetAssetAddr = '0x0000000000000000000000000000000000000000'
+    const balanceQueryAddrs = [mainnetAssetAddr]
+    for (const rawToken of Object.values(this.ethEngine.allTokensMap)) {
+      const token = asMaybeContractLocation(rawToken.networkLocation)
+      if (token != null) balanceQueryAddrs.unshift(token.contractAddress)
+    }
+
+    let funcs: Array<() => Promise<any>> = []
+    rpcServers.forEach(rpcServer => {
+      const rpcServerWithApiKey = this.addRpcApiKey(rpcServer)
+      const ethProvider = new ethers.providers.JsonRpcProvider(
+        rpcServerWithApiKey,
+        chainParams.chainId
+      )
+
+      const contract = new ethers.Contract(
+        ethBalCheckerContract,
+        ETH_BAL_CHECKER_ABI,
+        ethProvider
+      )
+
+      funcs.push(
+        async () =>
+          await contract.balances(
+            [walletLocalData.publicKey],
+            balanceQueryAddrs
+          )
+      )
+    })
+
+    // Randomize provider priority to distribute RPC provider load
+    funcs = shuffleArray(funcs)
+    const balances = await asyncWaterfall(funcs).catch(e => {
+      this.logError(
+        'checkEthBalChecker',
+        new Error(`All rpc servers failed eth balance checks: ${e}`)
+      )
+      return []
```

Comment:
Throw the error instead of returning an array. If this fails we shouldn't pretend that it didn't.